src/Domain/Creators/AlignedLattice.hpp

```diff
@@ -18,6 +19,51 @@
 namespace domain {
 namespace creators {
 
+template <size_t VolumeDim>
+struct RefinementRegion {
+  std::array<size_t, VolumeDim> lower_bound;
+  std::array<size_t, VolumeDim> upper_bound;
+  std::array<size_t, VolumeDim> refinement;
+
+  struct LowerBound {
```

Comment:
how about `LowerCornerIndex` and `UpperCornerIndex` to avoid possible confusion with coordinate values or Block indices